src/wallet/scriptpubkeyman.cpp

```diff
@@ -1087,16 +1087,19 @@ void LegacyScriptPubKeyMan::AddKeypoolPubkeyWithDB(const CPubKey& pubkey, const
     m_pool_key_to_index[pubkey.GetID()] = index;
 }
 
-void LegacyScriptPubKeyMan::KeepDestination(int64_t nIndex, const OutputType& type, const CPubKey& pubkey)
+void LegacyScriptPubKeyMan::KeepDestination(int64_t nIndex, const OutputType& type)
```

Comment:
In commit "Key pool: Change ReturnDestination interface to take address instead of key" (8e70f2b4150ca2d25f06cc65295cb27a86c49681)

This should call `m_reserved_key_to_index.erase(nIndex);` to clean up

Reply:
Fixed.